sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlanner.scala

```diff
@@ -30,9 +29,9 @@ import org.apache.spark.sql.execution.datasources.v2.DataSourceV2Strategy
 import org.apache.spark.sql.internal.SQLConf
 
 class SparkPlanner(val session: SparkSession, val experimentalMethods: ExperimentalMethods)
-  extends SparkStrategies with SQLConfHelper {
+  extends SparkStrategies {
 
-  override def conf: SQLConf = session.sessionState.conf
+  protected def conf: SQLConf = session.sessionState.conf
```

Comment:
This case looks somewhat different. Also, is it necessary to restrict the access scope of the function here?

Reply:
I think should restrict it.

Reply:
1. Are there any other similar cases? 
2. Since this case is somewhat different from the other cases in the PR, could we specifically mention it in the PR description?

Reply:
1. I only find these cases.
2. I have updated the PR description.

Reply:
Although `SparkPlanner` is a private API, it's meaningless to break the API (`def conf` becomes protected instead of public) for no good.

Reply:
@cloud-fan OK. Let me revert it back.